acceptance/analyzer_test.go

```diff
@@ -76,317 +77,496 @@ func TestAnalyzer(t *testing.T) {
+		h.WithFlags(
+			"--build-arg", fmt.Sprintf("cnb_platform_api=%s", api.Platform.Latest()),
+			"-f", filepath.Join(analyzeDockerContext, dockerfileName),
+		),
 	)
 	defer h.DockerImageRemove(t, analyzeImage)
 
-	spec.Run(t, "acceptance-analyzer", testAnalyzer, spec.Parallel(), spec.Report(report.Terminal{}))
+	for _, api := range api.Platform.Supported {
+		spec.Run(t, "acceptance-analyzer/"+api.String(), testAnalyzerBuilder(api.String()), spec.Parallel(), spec.Report(report.Terminal{}))
+	}
 }
 
-func testAnalyzer(t *testing.T, when spec.G, it spec.S) {
-	var copyDir, containerName, cacheVolume string
-
-	it.Before(func() {
-		containerName = "test-container-" + h.RandString(10)
-		var err error
-		copyDir, err = ioutil.TempDir("", "test-docker-copy-")
-		h.AssertNil(t, err)
-	})
-
-	it.After(func() {
-		if h.DockerContainerExists(t, containerName) {
-			h.Run(t, exec.Command("docker", "rm", containerName))
-		}
-		if h.DockerVolumeExists(t, cacheVolume) {
-			h.DockerVolumeRemove(t, cacheVolume)
-		}
-		os.RemoveAll(copyDir)
-	})
-
-	when("called without an app image", func() {
-		it("errors", func() {
-			cmd := exec.Command(
-				"docker", "run", "--rm",
-				"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
-				analyzeImage,
-				ctrPath(analyzerPath),
-			) // #nosec G204
-			output, err := cmd.CombinedOutput()
-
-			h.AssertNotNil(t, err)
-			expected := "failed to parse arguments: received 0 arguments, but expected 1"
-			h.AssertStringContains(t, string(output), expected)
+func testAnalyzerBuilder(apiString string) func(t *testing.T, when spec.G, it spec.S) {
+	return func(t *testing.T, when spec.G, it spec.S) {
+		var copyDir, containerName, cacheVolume string
+
+		it.Before(func() {
+			containerName = "test-container-" + h.RandString(10)
+			var err error
+			copyDir, err = ioutil.TempDir("", "test-docker-copy-")
+			h.AssertNil(t, err)
 		})
-	})
-
-	when("cache image tag and cache directory are both blank", func() {
-		it("warns", func() {
-			output := h.DockerRun(t,
-				analyzeImage,
-				h.WithFlags("--env", "CNB_PLATFORM_API="+latestPlatformAPI),
-				h.WithArgs(
-					ctrPath(analyzerPath),
-					"some-image",
-				),
-			)
 
-			expected := "Not restoring cached layer metadata, no cache flag specified."
-			h.AssertStringContains(t, output, expected)
+		it.After(func() {
+			if h.DockerContainerExists(t, containerName) {
+				h.Run(t, exec.Command("docker", "rm", containerName))
+			}
+			if h.DockerVolumeExists(t, cacheVolume) {
+				h.DockerVolumeRemove(t, cacheVolume)
+			}
+			os.RemoveAll(copyDir)
 		})
-	})
 
-	when("the provided layers directory isn't writeable", func() {
-		it("recursively chowns the directory", func() {
-			h.SkipIf(t, runtime.GOOS == "windows", "Not relevant on Windows")
+		when("called without an app image", func() {
+			it("errors", func() {
+				h.SkipIf(t, api.MustParse(apiString).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7")
```

Comment:
This is great!

Reply:
Could we make the message a bit more descriptive - something like "Platform API >= 0.7 does not accept an image argument"